app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt

```diff
@@ -54,14 +56,28 @@ sealed class Credential {
     }
 }
 
-class SignupViewModel(private val users: Users) : ViewModel() {
+class SignupViewModel(
+    private val users: Users,
+    private val checkForInternetUseCase: CheckForInternetUseCase
+) : ViewModel() {
 
     private val _state = MutableLiveData(SignUpState())
     val state: LiveData<SignUpState> = _state
 
+    private val _isInternetAvailable = MutableLiveData<Boolean>()
+    val isInternetAvailable: LiveData<Boolean> = _isInternetAvailable
```

Comment:
This is my mistake, I should I mentioned this earlier.

For the each screen, we're trying to keep it simple by having them all rely on a single 'state' object called "state". In this case we can put the 'isInternetAvailable' as a field inside the SignupState object. This object is passed to the screen already, so you'd access it like 'state.isInternetAvailble'.

To see how the state object is updated, you can see the 'updateName' method.